gprofiler/merge.py

```diff
@@ -118,7 +124,10 @@ def merge_perfs(perf_all: Iterable[Mapping[str, str]], process_perfs: Mapping[in
         if process_perf_count > 0:
             ratio = perf_all_count / process_perf_count
             for stack, count in process_stacks.items():
-                full_stack = ";".join([process_names[pid], stack])
-                new_samples[full_stack] += round(count * ratio)
-
-    return "\n".join((f"{stack} {count}" for stack, count in new_samples.items()))
+                stack_line = ";".join([docker_client.get_container_name(pid), process_names[pid], stack])
+                new_samples[stack_line] += round(count * ratio)
+    container_names = docker_client.container_names
+    docker_client.reset_cache()
+    output = [f"#{json.dumps({'containers': container_names, 'hostname': socket.gethostname()})}"]
```

Comment:
What's the hostname doing here? Let's discuss it in Slack

Reply:
@talsaiag and @ezean21 asked for it. The hostname will also show up in the filename that's saved to S3.